### ISSUE TITLE
Scaladoc examples for `NoArg`

### ISSUE THREAD

**Issue by jraman** [scalatest/scalatest#351]
In the scaladoc for [`NoArg`](http://doc.scalatest.org/2.2.0/index.html#org.scalatest.fixture.NoArg):
- method `evaluating` in trait Matchers is deprecated.
- The _"throw when made with the wrong name"_ test fails - it does not throw an Exception.
- In the modified 2.11 example, `NoArg` no longer needs to be imported.

It would be good to add an implementation for `MyActor` to make the example more complete.

**Comment by mkurz** [comment]
@cheeseng @bvenners I guess `NoArg` does not work anymore when using Scala 3 because it depends on `DelayedInit` whose "...special handling ... is no longer supported..."? See https://docs.scala-lang.org/scala3/reference/dropped-features/delayed-init.html

**Comment by cheeseng** [comment]
@mkurz Yes you're right, NoArg will not work in Scala 3, I submitted the following PR to update NoArg about this: 

https://github.com/scalatest/scalatest/pull/2228

Thanks for pointing it out!
